### PR TITLE
[py-pyprecice] add v2.4.0.0

### PR DESCRIPTION
**Pull request by IshaanDesai**
This PR adds the latest release v2.4.0.0.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pyprecice/package.py [hunk 1]
```diff
@@ -32,6 +33,7 @@ class PyPyprecice(PythonPackage):
     version("2.0.0.1", sha256="96eafdf421ec61ad6fcf0ab1d3cf210831a815272984c470b2aea57d4d0c9e0e")
 
     depends_on("precice@develop", when="@develop")
+    depends_on("precice@2.4.0", when="@2.4.0.1:2.4.0")
     depends_on("precice@2.3.0", when="@2.3.0.1:2.3.1")
     depends_on("precice@2.2.1", when="@2.2.1.1:2.2.1")
     depends_on("precice@2.2.0", when="@2.2.0.1:2.2.0")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyprecice/package.py:39]: Can you update the list of build deps below? Packaging is still required and a specific version of pip is needed:
```
requires = ["setuptools", "wheel", "Cython>=0.29", "packaging", "pip>=19.0.0", "numpy", "mpi4py"]
```

Comment (IshaanDesai) [reply var/spack/repos/builtin/packages/py-pyprecice/package.py:39]: I now added `19.0.0` as a lower bound to the pip dependency. I am a bit confused though, where is the line: 

```
requires = ["setuptools", "wheel", "Cython>=0.29", "packaging", "pip>=19.0.0", "numpy", "mpi4py"]
```
exactly? I did not understand where the version requirement for `pip` come from. Sorry, a bit new at this as I am taking this over from @ajaust

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-pyprecice/package.py:39]: These requirements come from `pyproject.toml`. See https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html for a basic introduction on Python's build system.

File: var/spack/repos/builtin/packages/py-pyprecice/package.py [hunk 2]
```diff
@@ -47,7 +49,7 @@ class PyPyprecice(PythonPackage):
     depends_on("py-mpi4py", type=("build", "run"))
     depends_on("py-cython@0.29:", type="build")
     depends_on("py-packaging", when="@:2.1", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyprecice/package.py]:
```suggestion
    depends_on("py-packaging", type="build")
```

File: var/spack/repos/builtin/packages/py-pyprecice/package.py [hunk 3]
```diff
@@ -47,7 +49,7 @@ class PyPyprecice(PythonPackage):
...
-    depends_on("py-pip", when="@:2.1", type="build")
+    depends_on("py-pip@19.0.0:", when="@:2.1", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyprecice/package.py]:
```suggestion
    depends_on("py-pip@19.0.0:", type="build")
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-pyprecice/package.py (hunk 1 above)
Comment (IshaanDesai) [reply var/spack/repos/builtin/packages/py-pyprecice/package.py:39]: Ah yes, the `pyproject.toml` file indeed. Thanks!

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
Doc test failures are unrelated, new Sphinx release is to blame. Will investigate later today.

**Pull request merged**